components/MyActivities.js

```diff
@@ -1,31 +1,130 @@
 import activities from '../sample-data/sample-activities.js';
 import MyActivityReview from './MyActivityReview.js';
+import { useState, useEffect } from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
 
 const MyActivities = () => {
 
+  const [show, setShow] = useState(false);
+  const [ reviewBody, setReviewBody ] = useState(null);
+  const [ reviewTitle, setReviewTitle ] = useState(null);
+
+  const handleClose = () => setShow(false);
+  const handleShow = (body, place) => {
+    setShow(true);
+    setReviewBody(body);
+    setReviewTitle(place);
+  }
+
+  let [allAct, setallAct] = useState([]);
+  let [rendered, setRendered] = useState([]);
+
+  var showMore = () => {
+      let al = allAct.length;
+      let rl = rendered.length;
+      if (al - rl <= 4) {
+          setRendered(allAct);
+      } else {
+          setRendered([
+              ...rendered,
+              allAct[rl],
+              allAct[rl+1],
+              allAct[rl+2],
+              allAct[rl+3]
+          ]);
+      }
+  }
+
+  var collapse = () => {
+      setRendered([
+          allAct[0],
+          allAct[1],
+          allAct[2],
+          allAct[3]
+      ]);
+  }
+
+  useEffect(() => {
+      setallAct(activities);
+      if (activities.length > 4) {
+          setRendered([
+              activities[0],
+              activities[1],
+              activities[2],
+              activities[3]
+          ]);
+      } else {
+          setRendered(activities);
+      }
+  }, [activities])
+
   return (
+      <div className="list-group" style={{backgroundColor: "#467850", width: "20rem"}}>
+          <h3 style={{textAlign: "center", color: "white"}}>My Activities</h3>
```

Comment:
Consider bringing in CSS values as variables instead of hard-coding it in-line.

When there are changes this is setting up for a lot of small pedantic changes